core/src/main/scala/io/projectglow/transformers/pipe/Piper.scala

```diff
@@ -191,6 +192,8 @@ class PipeIterator(
     try {
       inputFormatter.init(stream)
       input.foreach(inputFormatter.write)
+    } catch {
+      case e: Throwable => logger.warn(e.toString)
```

Comment:
You still want to throw the exception. Also, prefer `NonFatal` to `Throwable`

Reply:
Good call - I wrapped this in `WithUtils.withCloseable`